lib/facter/pe_status_check.rb

```diff
@@ -201,6 +201,19 @@
     { S0022: validity }
   end
 
+  chunk(:S0029) do
+    next unless PEStatusCheck.replica? || PEStatusCheck.postgres? || PEStatusCheck.primary?
+    # check if concurrnet connections to Postgres approaching 90% defined
+    approaching_limit = false
+    maximum = PEStatusCheck.max_connections.to_i
+    current = PEStatusCheck.cur_connections.to_i
+    percent_used = current / maximum * 100
+    if percent_used >= 90
+      approaching_limit = true
+      { S0029: approaching_limit }
```

Comment:
This line should be outside of the `if` condition, or else this chunk is not being set when false. So you end up in a state where `S0029` is missing when it is false. 

You can do a quick modification to fix this like below. 

```
if percent_used >= 90
      approaching_limit = true
end
 { S0029: approaching_limit }
```

However, an easier to read and write revision of this code is to use the `percent_used >= 90` condition in the returned hash like the following. 

```
chunk(:S0029) do
    next unless PEStatusCheck.replica? || PEStatusCheck.postgres? || PEStatusCheck.primary?
    # check if concurrnet connections to Postgres approaching 90% defined

    maximum = PEStatusCheck.max_connections.to_i
    current = PEStatusCheck.cur_connections.to_i
    percent_used = current / maximum * 100
    { S0029: percent_used >= 90 }
end
```

To add some error handling, you will want to ensure `maximum` is not 0 or catch the exception when it is. The code below just catches the `ZeroDivisionError` exception, logs a debug message, and sets the fact to false.

```
 chunk(:S0029) do
    next unless PEStatusCheck.replica? || PEStatusCheck.postgres? || PEStatusCheck.primary?
    # check if concurrnet connections to Postgres approaching 90% defined

    maximum = PEStatusCheck.max_connections.to_i
    current = PEStatusCheck.cur_connections.to_i
    percent_used = current / maximum * 100
    { S0029: percent_used >= 90 }
    rescue ZeroDivisionError
      Facter.debug('pe_status_check.S0029 failed to get max_connections')
      { S0029: false }
end
```